lib/logging/addon/components/logging/new-edit/component.js

```diff
@@ -1,82 +1,269 @@
+      const pageScope = this.get('pageScope');
+
+      set(this, 'testing', true);
+
+      let url = `${ pageScope }loggings?action=test`
+
+      if (get(this, 'targetType') === 'customTarget') {
+        url = `${ pageScope }loggings?action=dryRun`
       }
 
-      model.setProperties({
-        elasticsearchConfig:    null,
-        splunkConfig:           null,
-        kafkaConfig:            null,
-        syslogConfig:           null,
-        fluentForwarderConfig:  null,
+      return gs.rawRequest({
+        url,
+        method: 'POST',
+        data,
+      }).then(() => {
+        setProperties(this, {
+          testOk:  true,
+          errors:  null,
+        });
+      }).catch((xhr) => {
+        setProperties(this, {
+          testOk:  false,
+          errors:  [xhr.body.message || xhr.body.code],
+        });
+      }).finally(() => {
+        setProperties(this, {
+          tested:  true,
+          testing: false,
+          model:   clone,
+        })
+        setTimeout(() => {
+          set(this, 'tested', false);
+        }, 3000);
       });
+    },
+    cancel() {
+      setProperties(this, {
+        targetType:    get(this, 'preTargetType'),
+        pasteOrUpload: false,
+      })
+    },
 
-      if (targetType === 'none') {
-        this._super(cb);
+    showPaste() {
+      setProperties(this, {
+        preTargetType: get(this, 'targetType'),
+        targetType:    'customTarget',
+        pasteOrUpload: true,
+      })
+    },
+  },
+
+  targetTypeChange: observer('targetType', function() {
+    set(this, 'errors', [])
+  }),
+
+  headerLabel: computed('pageScope', function() {
+    const ps = get(this, 'pageScope');
 
-        return;
+    if (ps === 'cluster') {
+      return 'loggingPage.clusterHeader';
+    } else {
+      return 'loggingPage.projectHeader';
+    }
+  }),
+
+  isClusterLevel: computed('pageScope', function() {
+    return get(this, 'pageScope') === 'cluster';
+  }),
+
+  saveDisabled: computed('originalModel.{id,targetType}', 'targetType', 'pasteOrUpload', function() {
+    return get(this, 'originalModel.targetType') === 'none'
+      && get(this, 'targetType') === 'none'
+      && !get(this, 'pasteOrUpload');
+  }),
+
+  willSave() {
+    let targetType = get(this, 'targetType');
+    const pageScope = this.get('pageScope');
+    const model = get(this, 'model');
+    const intl = get(this, 'intl')
+
+    set(model, 'clusterId', get(this, 'cluster.id'));
+    if (pageScope === 'project') {
+      set(model, 'projectId', get(this, 'project.id'));
+    }
+
+    setProperties(model, {
+      elasticsearchConfig:    null,
+      splunkConfig:           null,
+      kafkaConfig:            null,
+      syslogConfig:           null,
+      fluentForwarderConfig:  null,
+      customTargetConfig:     null,
+    });
+
+    if (targetType === 'none') {
+      return true;
+    }
+
+    set(this, 'errors', [])
+    const errors = get(this, 'errors') || []
```

Comment:
If you're setting to errors to empty array on 169 why add the or? You could write this like 

```
const errors = set(this, 'errors', []);
```

Set returns the new property so this accomplishes the same thing in a cleaner manner.